lib/SILGen/SILGenPoly.cpp

```diff
@@ -490,6 +490,28 @@ ManagedValue Transform::transform(ManagedValue v,
     }
   }
 
+  // - upcasts for collections
```

Comment:
It looks like you should check that the input and output struct decls are the same. Otherwise, `[T]` to AnyHashable conversion will fail, since we check for that further down. Can you add a test?

Reply:
Sure, will do!